server/src/main/java/org/opensearch/gateway/remote/RemoteManifestManager.java

```diff
@@ -124,8 +125,10 @@ RemoteClusterStateManifestInfo uploadManifest(
                 .metadataVersion(clusterState.metadata().version())
                 .transientSettingsMetadata(uploadedMetadataResult.uploadedTransientSettingsMetadata)
                 .clusterStateCustomMetadataMap(uploadedMetadataResult.uploadedClusterStateCustomMetadataMap)
-                .hashesOfConsistentSettings(uploadedMetadataResult.uploadedHashesOfConsistentSettings);
+                .hashesOfConsistentSettings(uploadedMetadataResult.uploadedHashesOfConsistentSettings)
+                .checksum(clusterStateChecksum);
             final ClusterMetadataManifest manifest = manifestBuilder.build();
+            logger.trace("uploading manifest [{}]", manifest);
```

Comment:
nit: use lambda invocation.

Reply:
Also just want to check, do we have an toString in manifest?

Reply:
Yes, manifest uses `toXContent` for `toString` which is implemented.